src/sfProject.ts

```diff
@@ -760,3 +724,22 @@ export class SfProject {
       .map(([key]) => key);
   }
 }
+
+/** differentiate between the Base PackageDir (path, maybe default) and the Packaging version (package and maybe a LOT of other fields) by whether is has the `package` property */
+export const isPackagingDirectory = (packageDir: PackageDir): packageDir is PackagePackageDir =>
```

Comment:
the are exported to help consumers use the types safely.

You can tell from the amount of ` as ` in packagingLib that the types weren't helping.